x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/AllocateAction.java

```diff
@@ -78,18 +85,24 @@ public AllocateAction(Integer numberOfReplicas, Map<String, String> include, Map
             throw new IllegalArgumentException("[" + NUMBER_OF_REPLICAS_FIELD.getPreferredName() + "] must be >= 0");
         }
         this.numberOfReplicas = numberOfReplicas;
+        this.totalShardsPerNode = totalShardsPerNode;
```

Comment:
We should validate that the setting is >= -1 here

Reply:
Done.